### PR TITLE
python312Packages.mtcnn: 0.1.1 -> 1.0.0; fix build

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

This PR fixes both `mtcnn` and `retinaface`.

Changelog: https://github.com/ipazc/mtcnn/releases/tag/v1.0.0

cc @derdennisop

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361156`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.deepface</li>
    <li>python311Packages.deepface.dist</li>
    <li>python311Packages.mtcnn</li>
    <li>python311Packages.mtcnn.dist</li>
    <li>python311Packages.retinaface</li>
    <li>python311Packages.retinaface.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.deepface</li>
    <li>python311Packages.deepface.dist</li>
    <li>python311Packages.mtcnn</li>
    <li>python311Packages.mtcnn.dist</li>
    <li>python311Packages.retinaface</li>
    <li>python311Packages.retinaface.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.deepface</li>
    <li>python311Packages.deepface.dist</li>
    <li>python311Packages.mtcnn</li>
    <li>python311Packages.mtcnn.dist</li>
    <li>python311Packages.retinaface</li>
    <li>python311Packages.retinaface.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/mtcnn/default.nix [hunk 1]
```diff
@@ -2,50 +2,57 @@
...
-  nativeCheckInputs = [ pytestCheckHook ];
+  nativeCheckInputs = [
+    keras
+    pytestCheckHook
+  ];
 
   meta = {
     description = "MTCNN face detection implementation for TensorFlow";
     homepage = "https://github.com/ipazc/mtcnn";
+    changelog = "https://github.com/ipazc/mtcnn/releases/tag/v1.0.0";
```
Comment (natsukium) [inline pkgs/development/python-modules/mtcnn/default.nix:55]:
```suggestion
    changelog = "https://github.com/ipazc/mtcnn/releases/tag/v${version}";
```

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 361156`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.deepface</li>
    <li>python311Packages.deepface.dist</li>
    <li>python311Packages.mtcnn</li>
    <li>python311Packages.mtcnn.dist</li>
    <li>python311Packages.retinaface</li>
    <li>python311Packages.retinaface.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.deepface</li>
    <li>python311Packages.deepface.dist</li>
    <li>python311Packages.mtcnn</li>
    <li>python311Packages.mtcnn.dist</li>
    <li>python311Packages.retinaface</li>
    <li>python311Packages.retinaface.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.deepface</li>
    <li>python311Packages.deepface.dist</li>
    <li>python311Packages.mtcnn</li>
    <li>python311Packages.mtcnn.dist</li>
    <li>python311Packages.retinaface</li>
    <li>python311Packages.retinaface.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/mtcnn/default.nix [hunk 2]
```diff
@@ -2,50 +2,57 @@
...
 
-  dependencies = [
-    opencv-python
-    tensorflow
-  ];
+  dependencies =
+    [
+      joblib
+      lz4
+      tensorflow
```
Comment (natsukium) [inline pkgs/development/python-modules/mtcnn/default.nix:39]: tensorflow seems to be an optional dependency
https://github.com/ipazc/mtcnn/blob/117e9c6d1a4a740ac497f65026161658d28dab1e/setup.py#L41-L48

Comment (GaetanLepage) [reply pkgs/development/python-modules/mtcnn/default.nix]: Indeed, it seems to work fine without it.

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**